core/src/elements/kick_bomb.rs

```diff
@@ -137,7 +144,19 @@ fn update_idle_kick_bombs(
             commands.add(
                 move |mut idle: CompMut<IdleKickBomb>, mut lit: CompMut<LitKickBomb>| {
                     idle.remove(entity);
-                    lit.insert(entity, LitKickBomb { age: 0.0 });
+                    lit.insert(
+                        entity,
+                        LitKickBomb {
+                            arm_delay: Timer::new(
+                                Duration::from_secs_f32(arm_delay.as_secs_f32()),
+                                TimerMode::Once,
+                            ),
+                            fuse_time: Timer::new(
+                                Duration::from_secs_f32(fuse_time.as_secs_f32()),
+                                TimerMode::Once,
+                            ),
```

Comment:
This can be simplified slightly:

```suggestion
                            arm_delay: Timer::new(
                                arm_delay,
                                TimerMode::Once,
                            ),
                            fuse_time: Timer::new(
                                fuse_time,
                                TimerMode::Once,
                            ),
```

Reply:
Ok this has been simplified / updated. committing now